File: archive/gym_tracker_compose/app/src/main/java/com/example/gymtracker/data/datasource/LocalStorageDatasource.kt
package com.example.gymtracker.data.datasource

import android.content.Context
import com.example.gymtracker.domain.ports.StoragePort
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

/**
 * Adapter implementing [StoragePort] using SharedPreferences.
 */
class LocalStorageDatasource(
    private val context: Context
) : StoragePort {

    private val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    override suspend fun get(key: String): String? = withContext(Dispatchers.IO) {
        prefs.getString(key, null)
    }

    override suspend fun set(key: String, value: String) = withContext(Dispatchers.IO) {
        prefs.edit().putString(key, value).apply()
    }

    override suspend fun remove(key: String) = withContext(Dispatchers.IO) {
        prefs.edit().remove(key).apply()
    }

    companion object {
        private const val PREFS_NAME = "gym_tracker_prefs"
    }
}

File: archive/gym_tracker_flutter/lib/data/datasources/local_storage_datasource.dart
import 'package:shared_preferences/shared_preferences.dart';

import '../../domain/ports/storage_port.dart';

/// Adapter implementing [StoragePort] using SharedPreferences.
class LocalStorageDatasource implements StoragePort {
  LocalStorageDatasource(this._prefs);

  final SharedPreferences _prefs;

  @override
  Future<String?> get(String key) => Future.value(_prefs.getString(key));

  @override
  Future<void> set(String key, String value) => _prefs.setString(key, value);

  @override
  Future<void> remove(String key) => _prefs.remove(key);
}
